src/test/java/org/vafer/jdeb/DebMakerTestCase.java

```diff
@@ -232,4 +241,180 @@ public void visit(TarArchiveEntry entry, byte[] content) throws IOException {
+
+        assertTrue(packageControlFile.isValid());
+
+        final Map<String, TarArchiveEntry> filesInDeb = new HashMap<String, TarArchiveEntry>();
+
+        ArchiveWalker.walkData(deb, new ArchiveVisitor<TarArchiveEntry>() {
+            public void visit(TarArchiveEntry entry, byte[] content) throws IOException {
+                filesInDeb.put(entry.getName(), entry);
+            }
+        }, Compression.GZIP);
+
+        assertTrue("longname file wasn't found in the package", filesInDeb.containsKey(".//var/log/loooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongname/"));
+        assertTrue("short wasn't found in the package", filesInDeb.containsKey("./var/log/short"));
+        assertTrue("Cannot delete the file " + deb, deb.delete());
+    }
+
+    @Test
+    public void testLongLinkNamePosix() throws Exception {
+        String directoryName = "/var/log/" + longPathSuffix();
+        Mapper mapper = new PermMapper(-1, -1, "root", "root", -1, -1, 0, null);
+        String[] directories = { directoryName };
+        Mapper[] mappers = { mapper };
+        DataProducerPathTemplate dataProducer =  new DataProducerPathTemplate(directories, null, null, mappers);
+
+        DataProducer linkProducer = new DataProducerLink("/var/log/short", directoryName, true, null, null, null);
+
+        File deb = folder.newFile("file.deb");
+
+        List<DataProducer> producers = Arrays.asList(linkProducer, dataProducer);
+
+        DebMaker maker = new DebMaker(new NullConsole(), producers, null);
+        maker.setControl(new File(getClass().getResource("deb/control").toURI()));
+        maker.setTarLongFileMode("posix");
+        maker.setDeb(deb);
+        maker.setCompression(Compression.GZIP.toString());
+
+        BinaryPackageControlFile packageControlFile = maker.createDeb(Compression.GZIP);
+
+        assertTrue(packageControlFile.isValid());
+
+        final Map<String, TarArchiveEntry> filesInDeb = new HashMap<String, TarArchiveEntry>();
+
+        ArchiveWalker.walkData(deb, new ArchiveVisitor<TarArchiveEntry>() {
+            public void visit(TarArchiveEntry entry, byte[] content) throws IOException {
+                filesInDeb.put(entry.getName(), entry);
+            }
+        }, Compression.GZIP);
+
+        assertTrue("longname file wasn't found in the package", filesInDeb.containsKey(".//var/log/loooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongname/"));
+        assertTrue("short wasn't found in the package", filesInDeb.containsKey("./var/log/short"));
+        assertTrue("Cannot delete the file " + deb, deb.delete());
+    }
+
+    private String longPathSuffix() {
```

Comment:
I would generate a long path based on `numDirectories` being the number of directory levels and `nameLength` being the length of the file name. For now they could be just local variables. But testing a few combinations could be a could idea. Pseudo code as in:

```
test(longPath(numDirectories=2, nameLength=10))  == OK
test(longPath(numDirectories=5, nameLength=30))  == FAIL
test(longPath(numDirectories=5, nameLength=120)) == FAIL
```